tests/unit_tests/.gitignore

```diff
@@ -0,0 +1,4 @@
+test_driver*
+testdriver*
+example_test_driver*
+example_testdriver*
```

Comment:
This .gitignore should either move into example_test, or be made generic.  I'd suggest the second. E.g. ignore *_testdriver and use the convention that all testdriver programs are called that way (which would mean renaming example_testdriver2 to example2_testdriver). That way not every subdir will need it's own .gitignore.